src/gtc/passes/gtir_k_boundary.py

```diff
@@ -0,0 +1,76 @@
+import math
+from typing import Any, Dict, Tuple, Union, cast
+
+from eve import NodeVisitor
+from eve.utils import XIterable
+from gtc import gtir
+from gtc.common import LevelMarker
+
+
+def _iter_field_names(node: Union[gtir.Stencil, gtir.ParAssignStmt]) -> XIterable[gtir.FieldAccess]:
+    return node.iter_tree().if_isinstance(gtir.FieldDecl).getattr("name").unique()
+
+
+class KBoundaryVisitor(NodeVisitor):
+    """For every field compute the boundary in k, e.g. (2, -1) if [k_origin-2, k_origin+k_domain-1] is accessed."""
+
+    def visit_Stencil(self, node: gtir.Stencil, **kwargs: Any) -> Dict[str, Tuple[int, int]]:
+        field_boundaries = {name: (-math.inf, -math.inf) for name in _iter_field_names(node)}
+        for vloop in node.vertical_loops:
+            self.generic_visit(vloop.body, vloop=vloop, field_boundaries=field_boundaries, **kwargs)
+        # if there is no left or right boundary set to zero
+        for name, b in field_boundaries.items():
+            field_boundaries[name] = (
+                b[0] if b[0] != -math.inf else 0,
+                b[1] if b[1] != -math.inf else 0,
+            )
+        return cast(Dict[str, Tuple[int, int]], field_boundaries)
```

Comment:
I prefer to explicitly mark `cast` as a `typing` artifact without run-time effect by using the fully qualified name:
```suggestion
        return typing.cast(Dict[str, Tuple[int, int]], field_boundaries)
```
Anyway, an alternative way to deal with this would be to add the proper type hint to the `field_boundaries` declaration using a large integer (e.g. `sys.maxsize`) as sentinel value instead of `math.inf`)

Reply:
Done. I started to use `math.inf` instead of `sys.maxsize` after integers became unbounded in python 3. Frequently `math.inf` is only temporarily used in a computation and shouldn't leak, but when it unintentionally does I find it easier to catch these errors.